push_notifications/django_rest_api.py

```diff
@@ -0,0 +1,74 @@
+from rest_framework import serializers, permissions, viewsets
+from .models import APNSDevice, GCMDevice
+
+
+class IsOwner(permissions.BasePermission):
+    """
+    Custom permission to only allow owners of an object to see it.
+    """
+
+    def has_object_permission(self, request, view, obj):
+        # Only allow the user to see their own devices
+        return obj.user == request.user
+
+
+class APNSDeviceSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = APNSDevice
+        fields = ('name', 'device_id', 'registration_id')
+
+    read_only = ('date_created',)
+
+
+class GCMDeviceSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = GCMDevice
+        fields = ('name', 'device_id', 'registration_id')
+
+    read_only = ('date_created',)
+
+
+class DeviceViewSetMixin(object):
+    lookup_field = 'registration_id'
+    queryset = APNSDevice.objects.all()
```

Comment:
Is it right? GCM Also inherits from this, why does it only refer to APNSDevices?